provisioning/transport/http/src/http.ts

```diff
@@ -0,0 +1,271 @@
+              if (err) {
+                this._safeCallback(callback, err);
+                this._fsm.transition('idle');
+              } else if (this._isAssigning(body)) {
+                /* Codes_SRS_NODE_PROVISIONING_HTTP_18_028: [ If the operation status response contains a success status code with a 'status' that is 'Assigning', `register` shall fire an `operationStatus` event with the deserialized body and continue polling. ] */
+                this.emit('operationStatus', body);
+                this._fsm.transition('waiting_to_poll', callback);
+              } else if (this._isAssigned(body)) {
+                /* Codes_SRS_NODE_PROVISIONING_HTTP_18_027: [ If the operation status response contains a success status code with a 'status' of 'Assigned', `register` shall stop polling and call the `callback` with `err` == null and the body containing the deserialized body. ] */
+                callback(null, body);
+                this._fsm.transition('idle');
+              } else {
+                /* Codes_SRS_NODE_PROVISIONING_HTTP_18_030: [ If the operation status response has a success code with a 'status' that is not 'Assigned' or  'Assigning', `register` shall call the callback with a `SyntaxError` error and stop polling. ] */
+                this._safeCallback(callback, new SyntaxError('badly formed registration response'));
+                this._fsm.transition('idle');
+              }
+            });
+          },
+          register: (authorization, forceRegistration, body, callback) => {
+            /* Codes_SRS_NODE_PROVISIONING_HTTP_18_036: [ `register` shall call the `callback` with an `InvalidOperationError` if it is called while a previous registration is in progress. ] */
+            callback(new errors.InvalidOperationError('registration already in progress.'));
+          },
+          disconnect : (callback) => {
+            this._cancelCurrentOperation((err) => {
+              this._fsm.transition('idle');
+              this._safeCallback(callback);
+            });
+          }
+        }
+      }
+    });
+  }
+
+  register(authorization: string, forceRegistration: boolean, body: any, callback: ProvisioningResponseCallback): void {
+    /* Codes_SRS_NODE_PROVISIONING_HTTP_18_002: [ `register` shall throw a `ReferenceError` if `callback` is falsy. ] */
+    if (!callback) {
+      throw new ReferenceError();
+    }
+
+    this._fsm.handle('register', authorization, forceRegistration, body, callback);
+  }
+
+  disconnect(callback?: (err: Error) => void): void {
+    this._fsm.handle('disconnect', callback);
+  }
+
+  private _cancelCurrentOperation(callback: (err: Error) => void): void {
+    /* Codes_SRS_NODE_PROVISIONING_HTTP_18_034: [ disconnect will cancel any http operations that are in progress ] */
+    /* Codes_SRS_NODE_PROVISIONING_HTTP_18_035: [ disconnect will cause polling to cease ] */
+    if (this._currentHttpRequest != null) {
+      this._currentHttpRequest.abort();
```

Comment:
> this._currentHttpRequest.abort(); [](start = 5, length = 34)

what do we gain aborting the request? can't we just let it die and not care?

Reply:
You're right.  That also allows us to not leak the ClientRequest from the RestApiClient interface.

---
In reply to: [141483242](https://github.com/Azure/azure-iot-sdk-node/pull/93#discussion_r141483242) [](ancestors = 141483242)